pkg/reaper/deployment.go

```diff
@@ -193,6 +201,55 @@ func computeProbe(probeTemplate *corev1.Probe) *corev1.Probe {
 	return probe
 }
 
+func computeImage(containerImage *api.ContainerImage) string {
```

Comment:
Even if we wind up having separate `ContainerImage` types for Reaper and for Stargate there is a lot of duplication with the `computeImage`, `computeImagePolicy`, and `computeImagePullSecrets` functions. I believe the latter two are identical. Could we introduce some methods and interfaces to eliminate some or most of the duplication? Each of the `ContainerImage` structs could implement some interface(s), and the functions could be refactored to take the interface types as arguments.

Reply:
> Could we introduce some methods and interfaces to eliminate some or most of the duplication?

I gave this a try in fa9cf7d, let me know what you think.